docs/specs/component.md

```diff
@@ -114,8 +114,9 @@ There is leeway in the implementation of these events:
 
 #### BytesReceived
 
-*Sources* MUST emit a `BytesReceived` event immediately after receiving bytes
-from the upstream source and before the creation of a Vector event.
+*Sources* MUST emit a `BytesReceived` event immediately after receiving
+and (optionally) filtering bytes from the upstream source and before the
+creation of a Vector event.
```

Comment:
Per the conversation in https://github.com/vectordotdev/vector/pull/9637 should this also include something about not counting unprocessable bytes? For example, journald messages that fail to decode or incoming HTTP requests where the content is invalid?

Reply:
Well, I think that's the question I raised above. Should that be part of this PR or done as part of #9855 to avoid stalling this further.

Reply:
Ah, gotcha, I'm fine doing it as a follow-up if you prefer. I just think the end state should be that invalid or filtered bytes are not counted.